src/tokens/GmxGlpVault.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "BoringSolidity/libraries/BoringERC20.sol";
+import "BoringSolidity/ERC20.sol";
+import "BoringSolidity/BoringOwnable.sol";
+import "interfaces/IERC20Vault.sol";
+import {GmxGlpWrapperData} from "tokens/GmxGlpWrapper.sol";
+
+/// @dev Vault version of the GLP Wrapper, auto compound yield increase shares value similar to sSPELL
+/// storage is the same as the wrapper and allow to use the same GmxGlpRewardHandler implementation.
+contract GmxGlpVault is GmxGlpWrapperData, IERC20Vault {
```

Comment:
Do we want to be ERC4626 compatible here?

Reply:
we could also add that functionality through upgrading the rewardRouter I would assume